java_gen/pre-written/src/main/java/org/projectfloodlight/openflow/types/IPAddress.java

```diff
@@ -169,4 +183,26 @@ else if (address instanceof Inet6Address)
     public static IPAddress<?> fromInetAddress(@Nonnull InetAddress address) {
         return of(address);
     }
+
+    @Nonnull
+    protected <T extends InetAddress> T toInetAddressSubclassChecked(
```

Comment:
Given that we control the subclasses, and those are only IPv4 and IPv6, and each of those results in a byte array of the right length, and thus will convert to the right kind of InetAddress, I don't really see how a consumer of these APIs should ever trigger those exceptions.

The correctness of the implementation in `IPv[46]Address` should be validated via unit test.

There's really no point to add all this complexity just to add a little bit of info to an exception that should never occur when you use the API... Or am I overlooking something?